Wikipedia/Code/WMFFeedContentSource.m

```diff
@@ -335,15 +336,24 @@ - (void)saveGroupForNews:(NSArray<WMFFeedNewsStory *> *)news pageViews:(NSDictio
         [newsGroup updateContentPreviewWithContent:news];
     }
     newsGroup.isVisible = isVisible;
-    [self addNewsNotificationGroupForNewsGroup:newsGroup inManagedObjectContext:moc];
 }
 
-- (void)addNewsNotificationGroupForNewsGroup:(WMFContentGroup *)newsGroup inManagedObjectContext:(NSManagedObjectContext *)moc {
+-(void)createGroupForNotificationsInManagedObjectContext:(NSManagedObjectContext *)moc date:(NSDate *)date {
+    WMFContentGroup *notificationsGroup = [self notificationAnnouncementForDate:date inManagedObjectContext:moc];
+    
+    if (notificationsGroup == nil) {
+        notificationsGroup = [moc createGroupOfKind:WMFContentGroupKindNotification forDate:date withSiteURL:self.siteURL associatedContent:nil];
+    }
+}
+
+- (void)saveNotificationsGroupInManagedObjectContext:(NSManagedObjectContext *)moc date:(NSDate *)date {
     NSUserDefaults *userDefaults = [NSUserDefaults standardUserDefaults];
-    if (newsGroup && newsGroup.isVisible && ![userDefaults wmf_inTheNewsNotificationsEnabled] && ![userDefaults wmf_didShowNewsNotificationCardInFeed]) {
-        NSURL *URL = [WMFContentGroup notificationContentGroupURLWithLanguageVariantCode:self.siteURL.wmf_languageVariantCode];
-        [moc fetchOrCreateGroupForURL:URL ofKind:WMFContentGroupKindNotification forDate:newsGroup.date withSiteURL:self.siteURL associatedContent:nil customizationBlock:NULL];
+
+    if (userDefaults.wmf_shouldShowNotificationsExploreFeedCard && ![userDefaults wmf_didShowNewsNotificationCardInFeed] && self.fetcher.session.isAuthenticated ){
```

Comment:
Thanks for the deep dive into my PR! It shouldn't use the old card userDefaults flag. I guess repurposing the card led me to several assumptions I shouldn't be making. I also agree that keeping this card's logic in `WMFAnnouncementsContentSource` makes more sense. It also makes it easier for future maintenance since it's the same type of card and no longer tied to all the news logic. I cherry-picked the commits from your branch so we keep the commit history.